framework/auth/core.py

```diff
@@ -1228,6 +1227,11 @@ def merge_user(self, user):
             user_settings.merge(addon)
             user_settings.save()
 
+        # Disconnect signal to prevent emails being sent about being a new contributor when merging users
+        # be sure to reconnect it at the end of this code block. Import done here to prevent circular import error.
+        from website.project.signals import contributor_added
+        from website.project.views.contributor import notify_added_contributor
```

Comment:
This is good, but as discussed in-person, it might be helpful to have a context manager for temporarily disconnecting signals like this:

```python
with temporarily_disconnected_to(contributor_added, notify_added_contributor):
  # ...
```

As an added bonus, you might contribute this back to Blinker (note that Blinker has `temporarily_connected_to`).